File: rental-service/src/main/java/com/carrental/rental/messaging/RentalEvent.java
package com.carrental.rental.messaging;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

/**
 * L'événement publié dans RabbitMQ.
 * C'est ce message qui est envoyé dans la queue et consommé par vehicle-service.
 *
 * C'est une bonne pratique de créer une classe dédiée pour les events,
 * ça rend le contrat entre les microservices explicite et clair.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RentalEvent {
    private Long rentalId;
    private Long vehicleId;
    private String customerId;
    private String customerName;
    private LocalDate startDate;
    private LocalDate endDate;
    private String eventType; // "CONFIRMED" ou "CANCELLED"
}

File: vehicle-service/Messaging/RentalEvent.cs
namespace VehicleService.Messaging;

using System.Text.Json;

/// <summary>
/// Représentation de l'event reçu depuis rental-service via RabbitMQ.
/// Les noms des propriétés doivent correspondre exactement à RentalEvent.java
/// pour que la désérialisation JSON fonctionne correctement.
/// </summary>
public class RentalEvent
{
    public long RentalId { get; set; }
    public long VehicleId { get; set; }
    public string CustomerId { get; set; } = string.Empty;
    public string CustomerName { get; set; } = string.Empty;
    public JsonElement StartDate { get; set; }
    public JsonElement EndDate { get; set; }
    public string EventType { get; set; } = string.Empty; // "CONFIRMED" ou "CANCELLED"
}
